packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  let userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    userConfig = await userConfig();
+  }
+
+  if (typeof userConfig === 'object' && userConfig !== null && !Array.isArray(userConfig)) {
+    return userConfig;
+  }
+
+  throw new MetadataValidationError(
+    `Metadata store config returned an unkown value, should be an object: "${userConfig}"`
+  );
+}
+
+/**
+ * Get the static configuration file path, based on the metadata context.
+ * This uses any custom name provided, but swaps out the extension for `.json`.
+ */
+export function getStaticConfigFile({
+  projectDir,
+  metadataPath,
+}: Pick<MetadataContext, 'projectDir' | 'metadataPath'>): string {
+  const configFile = path.join(projectDir, metadataPath);
+  const configExtension = path.extname(configFile);
+
+  return path.join(projectDir, `${path.basename(configFile, configExtension)}.json`);
+}
+
+/**
+ * Load the store configuration from a metadata context.
+ * This can load `.json` and `.js` config files, using `require`.
+ * It throws MetadataValidationErrors when the file doesn't exist, or contains errors.
+ * The user is prompted to try anyway when errors are found.
+ */
+export async function loadConfigAsync({
```

Comment:
This function is util for reading local configuration, but during the error handling, you are assuming that it will be only called when pushing an update to store. This function should only return the config or throw an error, those error logs and prompt to continue should be separated.

Reply:
Fair point, will split it up.

Reply:
Done